src/network/serverpackethandler.cpp

```diff
@@ -488,8 +488,12 @@ void Server::process_PlayerPos(RemotePlayer *player, PlayerSAO *playersao,
 	pitch = modulo360f(pitch);
 	yaw = wrapDegrees_0_360(yaw);
 
-	playersao->setBasePosition(position);
-	player->setSpeed(speed);
+	if (!playersao->isAttached()) {
+		// Only update player positions when moving freely
+		// to not interfere with attachment handling
+		playersao->setBasePosition(position);
+		player->setSpeed(speed);
+	}
 	playersao->setLookPitch(pitch);
 	playersao->setPlayerYaw(yaw);
 	playersao->setFov(fov);
```

Comment:
the `if (playersao->checkMovementCheat()) {` down further too

Reply:
I'd like to keep this as-is in case the movement check is later used in another location.
https://github.com/minetest/minetest/pull/10952#discussion_r575823239

IMO deciding whether a check should performed is the task of the anticheat function, not its caller. At this point it's mainly a question about coding practices. I can adjust that if you insist.

Reply:
No that's fine.